src/main/java/persistence/sql/ddl/DefaultEntityManager.java

```diff
@@ -0,0 +1,194 @@
+        if (containsCache(clazz, id)) {
+            return (T) getCache(clazz, id);
+        }
+
+        EntityTable entityTable = EntityTable.from(clazz);
+
+        String query = selectDMLGenerator.generateFindById(entityTable, id);
+
+        List<Object> results = jdbcTemplate.query(query, toObject(clazz, id, entityTable));
+
+        if (results.isEmpty()) {
+            return null;
+        } else if (results.size() >= 2) {
+            throw new RuntimeException("Expected 1 result, got " + results.size());
+        }
+
+        return (T) results.get(0);
+    }
+
+    private boolean containsCache(Class clazz, Object id) {
+        Entry<Class, Object> entry = new SimpleEntry<>(clazz, id);
+
+        return cache.containsKey(entry);
+    }
+
+    private Object getCache(Class clazz, Object id) {
+        Entry<Class, Object> entry = new SimpleEntry<>(clazz, id);
+
+        return cache.get(entry);
+    }
+
+    @Override
+    public Object persist(Object entity) {
+        Class<?> clazz = entity.getClass();
+        EntityTable entityTable = EntityTable.from(clazz);
+
+        Field field = entityTable.getFieldByIdColumn();
+
+        Object id = FieldUtils.getValue(field, entity);
+
+        if (id == null) {
+            Object valueOfId = insert(entity, Statement.RETURN_GENERATED_KEYS);
+
+            FieldUtils.setValue(field, entity, valueOfId);
+
+            // id 가져오기
+            putCache(clazz, valueOfId, entity);
+        } else {
+            if (find(clazz, id) == null) {
+                insert(entity, Statement.NO_GENERATED_KEYS);
+            } else {
+                update(entity);
+            }
+
+            putCache(clazz, id, entity);
+        }
+
+        return entity;
+    }
+
+    private Object insert(Object entity, int autoGeneratedKeys) {
+        String queryOfInsert = insertDMLGenerator.generate(entity);
+
+        try (PreparedStatement statement = connection.prepareStatement(queryOfInsert, autoGeneratedKeys)) {
+            statement.execute();
+
+            if (autoGeneratedKeys == PreparedStatement.RETURN_GENERATED_KEYS) {
+                ResultSet generatedKeys = statement.getGeneratedKeys();
+
+                generatedKeys.next();
+
+                return generatedKeys.getObject(1);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+
+        return null;
+    }
+
+    private void update(Object entity) {
+        String query = updateDMLGenerator.generate(entity);
+
+        jdbcTemplate.execute(query);
+    }
+
+    @Override
+    public void remove(Object entity) {
+        Class<?> clazz = entity.getClass();
+        EntityTable entityTable = EntityTable.from(clazz);
+
+        Field field = entityTable.getFieldByIdColumn();
+
+        Object id = FieldUtils.getValue(field, entity);
+
+        removeCache(clazz, id);
+
+        String query = deleteDMLGenerator.generateDeleteById(entityTable, id);
```

Comment:
```suggestion
        EntityTable entityTable = EntityTable.from(entity);
        Object id = entityTable. extractId(entity);
        removeCache(clazz, id);
        String query = deleteDMLGenerator.generateDeleteById(entityTable, id);
```
위 코멘트를 반영하면 여기도 이런 코드로 변경될 수 있을 것 같아요 :)